src/openvic-simulation/types/Sliders.hpp

```diff
@@ -0,0 +1,67 @@
+#pragma once
+
+#include <cassert>
+#include <cstdint>
+
+#include "openvic-simulation/utility/Getters.hpp"
+
+namespace OpenVic {
+  // TODO: make struct for tariff slider, unlike other sliders, you can have negative values down to -100
+  struct slider_value_t {
+  using int_type = int32_t;
+  private:
+      int_type PROPERTY(min);
+      int_type PROPERTY(max);
+      int_type PROPERTY(value);
+      const int_type PROPERTY(strict_min);
+      const int_type PROPERTY(strict_max)
+      
+
+  public:
+      constexpr slider_value_t(int_type value = 50, int_type min = 0, int_type max = 100) : strict_min(0), strict_max(100) {
+        assert(min >= strict_min && min <= strict_max && min >= strict_min && min <= strict_max);
+        set_value(value);
+        set_bounds(max,min);
+      }
+      void set_value(int_type new_value) {
+        if (min <= max)
+          {
+            if (new_value > max) {
+              value = max;
+            }
+            else if (new_value < min) {
+              value = min;
+            }
+            else {
+              value = new_value;
+            }
+          }
+        // You *can* actually have min > max in Victoria 2
+        // Such a situation will result in only being able to be move between the max and min value. 
+        // This logic replicates this "feature"
+        else {
+          if (new_value > max) {
+            value = min;
+          }
+          else {
+            value = max;
+          }
+        }
+      }
+      void set_bounds(int_type new_max, int_type new_min) {
+        assert(min >= strict_min && min <= strict_max && min >= strict_min && min <= strict_max);
+        min = new_min;
+        max = new_max;
+        set_value(value);
+      }
+      slider_value_t& operator=(const slider_value_t& other) {
+          if (this != &other) {
+              min = other.min;
+              max = other.max;
+              value = other.value;
+
+          }
+          return *this;
+      }
+    };
+}
```

Comment:
```suggestion
#include <algorithm>
#include <cstdint>

#include "openvic-simulation/utility/Getters.hpp"

namespace OpenVic {
	using slider_value_int_type = int32_t;

	template<slider_value_int_type STRICT_MIN, slider_value_int_type STRICT_MAX>
	struct slider_value_t {
		using int_type = slider_value_int_type;

		static constexpr int_type strict_min = STRICT_MIN;
		static constexpr int_type strict_max = STRICT_MAX;

		static_assert(strict_min < strict_max);

	private:
		int_type PROPERTY(min);
		int_type PROPERTY(max);
		int_type PROPERTY(value);

	public:
		constexpr slider_value_t(
			int_type new_value = (strict_max - strict_min) / 2, int_type new_min = strict_min, int_type new_max = strict_max
		) {
			set_bounds(new_max, new_min);
			set_value(new_value);
		}

		constexpr void set_value(int_type new_value) {
			if (min <= max) {
				value = std::clamp(new_value, min, max);
			} else {
				// You *can* actually have min > max in Victoria 2
				// Such a situation will result in only being able to be move between the max and min value.
				// This logic replicates this "feature"
				if (new_value > max) {
					value = min;
				} else {
					value = max;
				}
			}
		}

		constexpr void set_bounds(int_type new_max, int_type new_min) {
			min = std::clamp(new_min, strict_min, strict_max);
			max = std::clamp(new_max, strict_min, strict_max);
			set_value(value);
		}

		constexpr slider_value_t& operator=(slider_value_t const& other) {
			if (this != &other) {
				min = other.min;
				max = other.max;
				value = other.value;
			}
			return *this;
		}
	};
}
```
There's quite a bit here but I think it will improve the struct a lot:
- Make strict min/max template arguments, allowing them to be customised for different slider values and also preventing issues with `operator=` between slider values with different strict min/max.
- You could add default values to the template arguments (0 and 100), but IMO it's more clear to explicitly define them where they're used (we won't have to do that every time as we can define other type aliases, as I'll show in `CountryInstance.hpp`).
- You can't use a type alias before it's defined, so we'll need to define `slider_value_int_type` before the template. By setting `int_type` inside the template to equal it, we make it accessible from the template's type while still only defining it in one place.
- Add variables for strict min/max so they can be accessed via instances of the template.
- The `static_assert(strict_min < strict_max);` will be checked during compilation, this is better than regular `assert`s which are only checked in debug builds (and will crash the program if they fail).
- In the constructor, we can set the default args based on the strict min/max template arguments, e.g. so that the tariff slider value has a default value of 0. We should call `set_bounds` before `set_value` as the bounds will start off default initialised to 0 (or a random value with MSVC which doesn't default initialise primitives unless you explicitly tell it to) and so setting the value first will bound it based on incorrect min/max. We could explicitly default initialise value/min/max, but just calling `set_bounds` then `set_value` is guarunteed to set everything up correctly.
- `set_value` and `set_bounds` need to be `constexpr` otherwise the constructor will fail to compile, and `operator=` can also be `constexpr`.
- `set_value` can use `std::clamp` for the `min <= max` case (this requires including `<algorithm>`) and `set_bounds` can also use it for both `min` and `max` as we ensured `strict_min < strict_max` with the earlier `static_assert`.

If you copy over the above code (keeping the `pragma once` above and making sure everything is using tabs rather than spaces) this file should be perfect.